File: Arrays/16/sol.java
import java.util.*;
import java.lang.*;
import java.io.*;
class GFG
 {
    static long []arr;
	public static void main (String[] args)throws Exception
	 {
	    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	    int test = Integer.parseInt(br.readLine());
	    while (test-- > 0)
	    {
	        int size = Integer.parseInt(br.readLine());
	        arr = new long[size];
	        String []buff = br.readLine().split("\\s+");
	        for (int i = 0; i < size; i++)
	        {
	            arr[i] = Integer.parseInt(buff[i]);
	            arr[i] *= arr[i];
	        }
	        if(get())
	        {
	            System.out.println("Yes");
	        }
	        else System.out.println("No");
	    }
	 }
	 static boolean get()
	 {
	     Arrays.sort(arr);
	     int size = arr.length;
	     for (int i =  size-1; i > 1; --i)
	     {
	         long sum = arr[i];
	         int low = 0,high = i-1;
	         while(low < high)
	         {
	             if(arr[low] + arr[high] == sum)return true;
	             if(arr[low]+arr[high] > sum )high--;
	             else low++;
	         }
	     }
	     return false;
	 }
}

File: Arrays/07/sol.py
def greaterThan(arr, val):
    low = 0
    total = len(arr)
    high = total - 1
    mid = 0
    res = total
    while  low <= high:
        mid = low + (high - low)//2
        if arr[mid] > val:
            res = mid 
            high = mid - 1
        else:
            low = mid + 1
    return total - res
def getTotal(arr1, arr2):
    arr1.sort()
    arr2.sort()
    #arr_1 = []
    arr_2 = []
    total = 0
    for i in range(4):
        #arr_1.append(arr1.count(i))
        arr_2.append(arr2.count(i))
    #print(arr_2)
    for i in arr1:
        #print(total)
        if i == 0:
            total += 0
        elif i == 1:
            total = total  + arr_2[0]
        elif i == 2:
            total = total + arr_2[0] + arr_2[1] + greaterThan(arr2,4)
        elif i == 3:
            total = total + arr_2[0] + arr_2[1] + arr_2[2] + greaterThan(arr2, i)
        elif i == 4:
            total = total + arr_2[0] + arr_2[1] + greaterThan(arr2, i)
        else:
            total = total + arr_2[0] + arr_2[1] + greaterThan(arr2, i)
    return total




for test in range(int(input())):
    buff = input()
    arr1 = [int(x) for x in input().split()]
    arr2 = [int(x) for x in input().split()]
    print(getTotal(arr1,arr2))
